app/marketing/management/commands/send_tax_report.py

```diff
@@ -0,0 +1,342 @@
+        if CODE in location_components:
+            code = location_components[CODE]
+            if location_temp:
+                location_temp += ', ' + code
+            else:
+                location_temp += code
+            # check if the worker is US based
+            if code == US:
+                us_worker = True
+    elif profile.locations:
+        location_components = profile.locations[-1]
+        if CITY in location_components:
+            location_temp += location_components[CITY]
+        if COUNTRY_NAME in location_components:
+            country_name = location_components[COUNTRY_NAME]
+            if location_temp:
+                location_temp += ', ' + country_name
+            else:
+                location_temp += country_name
+        if COUNTRY_CODE in location_components:
+            country_code = location_components[COUNTRY_CODE]
+            if location_temp:
+                location_temp += ', ' + country_code
+            else:
+                location_temp += country_code
+            if location_components[COUNTRY_CODE] == US:
+                us_worker = True
+    if location_temp:
+        location = location_temp
+    return location, us_worker
+
+
+def zip_dir(username, username_path):
+    zip_file_path = os.path.join(username_path, username) + '.zip'
+    zipf = zipfile.ZipFile(zip_file_path, 'w', zipfile.ZIP_DEFLATED)
+    for root, dirs, files in os.walk(username_path):
+        for filename in files:
+            if filename.endswith('.zip'):
+                continue
+            abs_path = os.path.join(root, filename)
+            zipf.write(abs_path, abs_path.split('tax_report')[1])
+    zipf.close()
+    return zip_file_path
+
+
+class Command(BaseCommand):
+
+    help = 'the tax report for last year'
+
+    def handle(self, *args, **options):
+        to_emails = []
+        zip_paths = []
+        profiles = Profile.objects.all()
+        tax_path = os.path.join(os.getcwd(), TAX_REPORT_PATH)
+        if not tax_path:
+            os.makedirs(tax_path)
+        b_source_type = ContentType.objects.get(app_label='dashboard', model='bountyfulfillment')
+        t_source_type = ContentType.objects.get(app_label='dashboard', model='tip')
+        g_source_type = ContentType.objects.get(app_label='grants', model='contribution')
+        for p in profiles:
+            csv_record = []
+            us_workers = {}
+            # Total Earning
+            earnings = Earning.objects.filter(
+                                            from_profile=p, 
+                                            network=WEB3_NETWORK, 
+                                            created_on__year=TAX_YEAR, 
+                                            value_usd__gt=0
+                                            ).exclude(to_profile=p)
+            # Bounties
+            # source type id 46 for bounties
+            bounties_fulfillments = earnings.filter(source_type_id=b_source_type.id)
```

Comment:
IMHO its much easier to just use the `Earnings` model than to iterate through all these different transaction types and have custom logic for each of them..

Reply:
Yes i could implement it without problems but depends when it has to be ready, i should change some parts of code and retesting it.

Reply:
Done